tests/atomic_ref/atomic_ref_common.h

```diff
@@ -0,0 +1,206 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common functions for the atomic_ref tests.
+//
+*******************************************************************************/
+
+#ifndef SYCL_CTS_ATOMIC_REF_COMMON_H
+#define SYCL_CTS_ATOMIC_REF_COMMON_H
+
+#include "../common/common.h"
+#include "../common/section_name_builder.h"
+#include "../common/type_coverage.h"
+
+#define SYCL_CTS_ATOMIC_REF_ON_HOST 0
+
+namespace atomic_ref_tests_common {
+using namespace sycl_cts;
+
+constexpr int expected_val = 42;
+constexpr int changed_val = 1;
+
+/**
```

Comment:
Yes, we use similar functions in different tests groups like: `get_full_conformance_type_pack()`, `get_lightweight_type_pack()`, `get_conformance_type_pack()`. However usually different groups of tests should be tested with different sets of types so it's not possible to make a generic function for these purposes. For example according to specification `sycl::atomic_ref` and `sycl::accessor` has different permitted types and because of it implementations of `get_lightweight_type_pack()` function are different for them:
* implementation in `accessor_common.h`
`inline auto get_lightweight_type_pack() {
  static const auto types =
      named_type_pack<bool, int, float>::generate("bool", "int", "float");
  return types;
}`

* implementation in `atomic_ref_common.h`
`
inline auto get_lightweight_type_pack() {
  static const auto types =
      named_type_pack<int, float>::generate("int", "float");
  return types;
}`

`get_conformance_type_pack()` have the same implementation in both files but it call different implementation of `get_full_conformance_type_pack()`, `get_lightweight_type_pack()`. 
It seems correct to create such functions for different groups of tests, since they are created in a common file in the directory of a particular test group and usually have different implementations. This makes it logical and convenient to analyze the code of a particular test group, since the code is located in the appropriate directory.

Reply:
Yes, but probably the core features could be consolidated, even if they do not have the same implementations today and be made configurable to be used in various places.
@bader I guess we do not have time to remove all these technical debt time bombs right now but we need a scoreboard somewhere to track all these.

Reply:
The obvious solution is to open a GitHub issue. We can use labels to separate technical debt issues from the rest if needed. Does it sound okay?

Reply:
That sounds good. Thanks.